spring-boot-project/spring-boot/src/test/java/org/springframework/boot/context/config/ConfigFileApplicationListenerTests.java

```diff
@@ -1150,6 +1151,46 @@ void locationsWithWildcardFilesShouldIgnoreHiddenDirectories() {
 		assertThat(this.environment.getProperty("fourth.property")).isNull();
 	}
 
+	@Test
+	void additionalProfilesCanBeIncludedFromProgrammaticallySetting() {
+		// gh-25704
+		SpringApplication application = new SpringApplication(Config.class);
+		application.setWebApplicationType(WebApplicationType.NONE);
+		application.setAdditionalProfiles("dev");
+		this.context = application.run();
+		// Active profile should win over default
+		assertThat(this.context.getEnvironment().getProperty("my.property")).isEqualTo("fromdevpropertiesfile");
+	}
+
+	@Test
+	void twoAdditionalProfilesCanBeIncludedFromProgrammaticallySetting() {
+		// gh-25704
+		SpringApplication application = new SpringApplication(Config.class);
+		application.setWebApplicationType(WebApplicationType.NONE);
+		application.setAdditionalProfiles("other", "dev");
+		this.context = application.run();
+		assertThat(this.context.getEnvironment().getProperty("my.property")).isEqualTo("fromdevpropertiesfile");
```

Comment:
I only want to test when there are many additional profiles rather than one.
In this test case, I want to check that which additional profile should win.

Maybe this test isn't related to the bug. Could I add this test to another PR?

Reply:
In that case, we don't need the test for the single additional profile and we can keep this one.

Reply:
@mbhave 
> In that case, we don't need the test for the single additional profile and we can keep this one.

You don't need the test for the multiple additional profiles, right?
If so, I will delete this test case.